### PR TITLE
profile page with login auth

### PR DESCRIPTION
**Pull request by RandInaim**
WIP for profile page and its authentication

### CODE REVIEW

File: src/server/router.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
 const { homeHandler, publicHandler, cuisineHandler, addRestaurantHandler, errorHandler } = require('./handler');
 const router = (request, response) => {
 	const endpoint = request.url;
-	console.log("hi", endpoint);
-	console.log("there", request);
+	console.log('hi', endpoint);
```
Comment (jema28) [inline src/server/router.js]: Please delete comments before committing ❤️

File: src/server/router.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+	console.log('there', request);
```
Comment (jema28) [inline src/server/router.js]: Same here.

File: src/server/handler.js [hunk 3]
```diff
@@ -1,107 +1,122 @@
...
+			'Content-Type': 'application/json'
 		});
 		response.end(dynamicData);
 	});
 };
 
 const addRestaurantHandler = (req, res) => {
-	let body = "";
-	req.on("data", chunk => {
+	let body = '';
```
Comment (jema28) [inline src/server/handler.js]: Okay, you need to sort out your linting to make it consistent for everyone (same config for everyone). Either pick single or double quotes.

File: src/server/handler.js [hunk 4]
```diff
@@ -1,107 +1,122 @@
...
-	const filePath = path.join(__dirname, "..", "..", "public", "index.html");
+const cookie = require('cookie');
+const jwt = require('jsonwebtoken');
+
+const { parse } = require('url');
+
+const homeHandler = (page, request, response) => {
+	let filePath = path.join(__dirname, '..', '..', 'public', `${page}.html`);
+	if (page === 'profile') {
+		console.log('cookies: ', request.headers.cookie);
```
Comment (jema28) [inline src/server/handler.js]: These are good for development but shouldn't be committed.

Comment (RandInaim) [reply src/server/handler.js]: do you mean the console log or these 4 lines?!

Comment (jema28) [reply src/server/handler.js]: The console logs 😸

File: src/server/handler.js [hunk 5]
```diff
@@ -1,107 +1,118 @@
...
+const cookie = require('cookie');
+const jwt = require('jsonwebtoken');
+
+const { parse } = require('url');
+
+const homeHandler = (page, request, response) => {
+	let filePath = path.join(__dirname, '..', '..', 'public', `${page}.html`);
+	if (page === 'profile') {
+		const comingToken = cookie.parse(request.headers.cookie).token;
+		const checkToken = jwt.verify(comingToken, '123456');
```
Comment (jema28) [inline src/server/handler.js]: This is an async function, it should take a callback.

**Pull request merged**